### PR TITLE
Application branch

### PR DESCRIPTION
**Pull request by oniso20**
The application branch focuses on the working application without styling.

The data comes from the json-server and you will need to run the code below to get it running before starting the react app.

```shell
json-server ./data/db.json —port 3000
```
The shell command gets the data serving from port 3000 which is the default.

## Key components
- Grid
- Row
- Nav
- OptionsCard

The grid and row replace the card.jsx as it helps me pass the colour styles to the correct cards easily.
The OptionsCard has a lot going on. It generates the answer options (keypad) based on the facts available.

## Hook
All functions are working from a single hook file. I have some notes in the hook file to explain most of the functions.

### CODE REVIEW

File: src/components/OptionsCard.jsx [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
+      <div className="options-container">
+        {facts &&
+          facts.map(({ id, text, correctPosition }) => {
+            const color = usedKeys[text];
+            return (
+              <div
+                className={color}
+                id="card"
+                key={text}
+                onClick={function (event) {
```
Comment (corbfon) [inline src/components/OptionsCard.jsx:34]: Your other `onClick` handlers use arrow functions (`() => {}`) - is there a reason this one is declared using the `function` keyword?

File: src/components/Row.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {[...Array(5 - facts.length)].map((_, idx) => (
+          <div key={idx}></div>
+        ))}
+      </div>
+    );
+  }
+
+  return (
+    <div className="rows">
+      <div></div>
```
Comment (corbfon) [inline src/components/Row.jsx:35]: These should probably get a class name so that they're both more descriptive and more extensible in the future

**Review by corbfon — CHANGES_REQUESTED** [review]
Hey Onis, nice work getting hooks together and setting up the initial components.

I'd like to see:

- A script included in the `scripts` section of `package.json` which will start the json server (you may also make it start both the json server and the app simultaneously if you'd like)
- Steps to start the project included in `README.md`
- The app shouldn't allow you to put the same answer in multiple times on the same row
- Your choice to do now or later, but we'll need to remove code that's commented out or `console.log`'s

DM me when changes are made, and I'm looking forward to seeing the style updates!

**Comment by oniso20** [comment]
Hello Corbin,

Thanks a lot for the extensive review, I'll make the needed changes and
revert back to you.

Best regards,
Onis Emem.

File: src/components/OptionsCard.jsx (hunk 1 above)
Comment (oniso20) [reply src/components/OptionsCard.jsx]: Oh... nice catch, I wasn't sure how to use 1 `onClick` handler for multiple events so I used it directly as seen from my search. I've made the update now.

File: src/components/Row.jsx (hunk 2 above)
Comment (oniso20) [reply src/components/Row.jsx:35]: Aha... I thought it will be more dry since I didn't have any major use for them

**Comment by oniso20** [comment]
Hi Corbin,

I have completed a major part of the styles and accounted for the same answer multiple times on the same row.

Just the scripts to conclude this phase.

Kind regards,
Onis Emem.

**Comment by oniso20** [comment]
Hi Corbin,

The Script in `package.json` and `README.md` has been successfully updated now #2

**Review by corbfon — APPROVED** [review]

**Pull request merged**